dolphinscheduler-bom/pom.xml

```diff
@@ -56,7 +56,7 @@
         <hadoop.version>3.2.4</hadoop.version>
         <cron-utils.version>9.1.6</cron-utils.version>
         <h2.version>2.2.220</h2.version>
-        <mysql-connector.version>8.0.16</mysql-connector.version>
+        <mysql-connector.version>8.0.33</mysql-connector.version>
```

Comment:
This is a significant change for users. We should disscuss it. @ruanwenjun @zhongjiajie @EricGao888

Reply:
it seem 8.0.33 is compatible with 8.0.16, But I want to know is why should we upgrade the version of mysql jdbc? I think we can fix this issue without this change.

Reply:
The upgrade is to ensure that the fields in the database are null and the objects are also null, rather than enumerating 0

Reply:
The enum are parsed by Mybatis-plus, I don't think this should be related to MySQL driver, is there any related doc?

BTW, if there is a reason to upgrade the MySQL driver, it's needed to use a single PR.

Reply:
This bug is caused by both the MySQL driver and mybatisplus. The driver returned 0 when the database field was null, and mybatisplus did not correctly use `wasnull` to determine. You can refer to this link specifically: https://github.com/baomidou/mybatis-plus/issues/5266

Reply:
This should be a serious bug, MySQL stores null but the driver returns 0.

I find the bug is `ResultSetImpl.getObject()` will return a non null value.
https://dev.mysql.com/doc/relnotes/connector-j/8.0/en/news-8-0-17.html